gp-advanced-select/gpadvs-enable-add-new-option.php

```diff
@@ -155,6 +160,15 @@ public function allow_created_choices_in_save_and_continue( $form, $ajax, $field
 		return $form;
 	}
 
+	public function disable_state_validation_for_advanced_select_field( $form ) {
+		foreach ( $form['fields'] as &$field ) {
+			if ( $field->id == $this->_args['field_id'] ) {
```

Comment:
I think this logic is a bit too naïve as it won't work if you enable it for a form globally.

It might be time to move the following into an `is_applicable_field` method:

```
if ( ! gp_advanced_select()->is_advanced_select_field( $field ) ) {
    continue;
}

// Check if this instance is targeting all Advanced Select fields or a specific field.
if ( ! empty( $this->_args['field_id'] ) && $this->_args['field_id'] != $field->id ) {
    continue;
}
```

Source: https://github.com/gravitywiz/snippet-library/blob/4a6204341ee37244ad8fe6329c137afa564f8283/gp-advanced-select/gpadvs-enable-add-new-option.php#L128-L135

Reply:
@claygriffiths  Thanks for pointing out the mistake of tapping the form globally. I have updated as per your suggestions, also refactored to have a new method `is_applicable_field`